feat: add mail component (#178)
instill-ai/component: application/email/v0/.compogen/extra-intro.mdx

You can connect different email server through Email component.

> We use SMTP protocol to send emails, So, you can set the SMTP server address and port in the component configuration.
> We use IMAP protocol to fetch emails, So, you can set the IMAP server address and port in the component configuration.

Reviewer: `s/So/so`

As a small suggestion, let's try to avoid repetition:
```suggestion
Emails are fetched and sent using the IMAP and SMTP protocols, respectively. You can set the server address and port for each protocol in the component configuration.
```
